summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -0,0 +1,103 @@
+package com.twitter.summingbird.storm.collector
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.online.option.SummerBuilder
+import backtype.storm.spout.SpoutOutputCollector
+import com.twitter.algebird.util.summer.AsyncSummer
+import com.twitter.util.{ Await, Future }
+import java.util.{ List => JList }
+import scala.collection.mutable.{ Map => MMap }
+import scala.collection.mutable.{ MutableList => MList }
+import scala.collection.{ Map => CMap }
+import scala.collection.JavaConverters._
+import java.util.{ List => JList }
+
+class AggregatorOutputCollector[K, V: Semigroup](in: SpoutOutputCollector, func: JList[AnyRef] => JList[AnyRef], summerBuilder: SummerBuilder, summerShards: KeyValueShards) extends TransformingOutputCollector(in, func) {
+
+  var spoutCaches = MMap[String, AsyncSummer[(K, V), Map[K, V]]]()
```

Comment:
Actually I think I would do this differently, and just roll the stream ID into the summing key, i.e.:

`AsyncSummer[((K, String), V), Map[(K, String), V]]`

This way, the memory controls, etc., don't depend on the number of streams being used.